test/src/e2e_vm_tests/test_programs/should_pass/stdlib/ec_recover_test/src/main.sw

```diff
@@ -33,13 +33,42 @@ fn main() -> bool {
     let signature: B512 = ~B512::from(sig_hi, sig_lo);
 
     // recover the address:
-    let recovered_address: Address = ec_recover_address(signature, msg_hash);
-    assert(recovered_address.value == address.value);
+    let address_result: Result<Address, EcRecoverError> = ec_recover_address(signature, msg_hash);
+    if let Result::Ok(a) = address_result {
+        assert(a.value == address.value);
+    } else {
+        panic(0);
+    };
 
     // recover the public key:
-    let recovered_pubkey: B512 = ec_recover(signature, msg_hash);
-    assert((recovered_pubkey.bytes)[0] == (pubkey.bytes)[0]);
-    assert((recovered_pubkey.bytes)[1] == (pubkey.bytes)[1]);
+    let pubkey_result: Result<B512, EcRecoverError> = ec_recover(signature, msg_hash);
+    if let Result::Ok(p) = pubkey_result {
+        assert(p == pubkey);
+    } else {
+        panic(0);
+    };
+
+    /////////////////////////////////////////
+    ///  Failure to recover
+    /////////////////////////////////////////
+
+    // using invalid data here to test the handling of failed pubkey/address recovery.
+    let bad_sig_hi = 0x000000000_8d8fe7dd522d88_000000000000000_34b6326ff51129776_000000000;
+    let bad_sig_lo = 0x000000000_4a11e49a2ee69f_000000000000000_dba8d779d323ab2a5_000000000;
+    let bad_signature: B512 = ~B512::from(bad_sig_hi, bad_sig_lo);
+
+    // this should return a Result::Err, so if it returns Result::Ok, we panic.
+    let pubkey_result: Result<B512, EcRecoverError> = ec_recover(bad_signature, msg_hash);
+    if let Result::Ok(v) = pubkey_result {
+        panic(0)
+    };
```

Comment:
Could you replace this with

```suggestion
    ec_recover(bad_signature, msg_hash).unwrap();
```

?

Reply:
No.
First, we can't chain methods like that yet. This would work syntax-wise:
```
let pubkey_result: Result<Address, EcRecoverError> = ec_recover_address(bad_signature, msg_hash);
pubkey_result.unwrap();
```
But, `unwrap()` will panic if the`Err` variant of Result is returned, but here I want to panic if the `Ok` variant is reurned.

Reply:
`ec_recover(bad_signature, msg_hash).unwrap();` is syntactically fine, if it isn't working then that's a bug.  If it's a type problem then a turbofish should handle that (or if not, bug)...

Reply:
ahhhhh yes, the turbofish. trying that.

Reply:
@otrho what's the turbofish syntax for this specific example?
I mean, I know it's `::<>`, but I want to make sure I'm using it correctly here before filing a bug report.
I've tried:
`ec_recover::<Result<B512, EcRecoverError>>(bad_signature, msg_hash).unwrap();`
and
`ec_recover(bad_signature, msg_hash).unwrap::<Result<B512, EcRecoverError>>();`

but I always get the error:
`Error parsing input: expected one of [op]` on the `unwrap()`

Reply:
I think the first one should've worked, although with our current parser you might need to wrap everything before the `.unwrap()` in parens.

The second one has the wrong type, since `unwrap()` would return the `B512`.

Reply:
Ok, so this works: `(ec_recover(bad_signature, msg_hash)).unwrap();`, so it's not a bug.
That said, I don't actually want to use `unwrap()` here for the reason I mentioned above.

Reply:
Oh I see. Then maybe use `.is_ok()`?

Reply:
>Then maybe use .is_ok()?
Actually, I need to use `.is_err()` here, but good idea, cleans it up nicely.
(I'm checking here that the `err` register has indeed been set to `1`, triggering the return of the `Err` variant.